src/middlewared/middlewared/plugins/pool_/attach_disk.py

```diff
@@ -54,7 +54,7 @@ async def attach(self, job, oid, options):
         verrors.check()
 
         guid = vdev['guid'] if vdev['type'] in ['DISK', 'RAIDZ1', 'RAIDZ2', 'RAIDZ3'] else vdev['children'][0]['guid']
-        disks = {options['new_disk']: {'create_swap': topology_type == 'data', 'vdev': []}}
+        disks = {options['new_disk']: {'create_swap': False}}
```

Comment:
Should we still expose `create_swap` considering we don't want to do that anymore ?

Reply:
For this PR, since the changes are so involved already, I didn't want to remove the argument. I wanted to do that in another PR separately.